webapp/src/components/link_tooltip/link_tooltip.jsx

```diff
@@ -0,0 +1,191 @@
+import React, {useEffect, useState} from 'react';
+import PropTypes from 'prop-types';
+import ReactMarkdown from 'react-markdown';
+import Octicon, {GitMerge, GitPullRequest, IssueClosed, IssueOpened} from '@primer/octicons-react';
+
+import Client from '../../client';
+import {validateGitlabURL} from '../../utils/regex_utils';
+import {isValidUrl} from '../../utils/url_utils';
+import './tooltip.css';
+
+const STATE_COLOR_MAP = {
+    OPENED_COLOR: '#28a745',
+    CLOSED_COLOR: '#cb2431',
+    MERGED_COLOR: '#6f42c1',
+};
+
+const STATE_TYPES = {
+    OPENED: 'opened',
+    CLOSED: 'closed',
+};
+
+const LINK_TYPES = {
+    MERGE_REQUESTS: 'merge_requests',
+    ISSUES: 'issues',
+};
+
+export const LinkTooltip = ({href, connected, gitlabURL}) => {
+    const [data, setData] = useState(null);
+    useEffect(() => {
+        if (!isValidUrl(href)) {
+            return;
+        }
+
+        const url = new URL(href);
+        const init = async () => {
+            if (url.origin === gitlabURL && validateGitlabURL(href)) {
+                const [owner, repo, , type, number] = href.split(`${url.hostname}/`)[1].split('/');
+                let res;
+                switch (type) {
+                case LINK_TYPES.ISSUES:
+                    res = await Client.getIssue(owner, repo, number);
+                    break;
+                case LINK_TYPES.MERGE_REQUESTS:
+                    res = await Client.getPullRequest(owner, repo, number);
+                    break;
+                }
+                if (res) {
+                    res = {...res, owner, repo, type};
+                }
+                setData(res);
+            }
+        };
+
+        if (data) {
+            return;
+        }
+
+        if (connected) {
+            init();
+        }
+    }, []);
+
+    const getIconElement = () => {
+        let color;
+        let iconType;
+        const {OPENED_COLOR, CLOSED_COLOR, MERGED_COLOR} = STATE_COLOR_MAP;
+        switch (data.type) {
+        case LINK_TYPES.MERGE_REQUESTS:
```

Comment:
Again, is it possible for this to be a different value?

Reply:
For now, we are showing the tooltip on the issue and merge request links. Is there any other type you want to support?

Reply:
@raghavaggarwal2308 I'm just wondering if it's possible that `data.type` is a different value than these two values. We'll want to handle the `default` case to take that into account